### DISCUSSION TITLE
Contributing Faiss on Apple platform (ARM+x86)

### DISCUSSION THREAD

**Discussion by spicymatt** [facebookresearch/faiss#1728] [General]
hello
I looked around the discussions and issues to see where you stand regarding the Apple platform. My understanding is that the 2 following points are still on the wishlist of some users (including myself):

1. using Apple's **Accelerate** framework for BLAS and LAPACK (to avoid some rejections on AppStore)
2. easily compiling Faiss for Apple systems (with support for **fat** libraries arm64+x86) 

I am willing to contribute to the library on these 2 aspects but, per your recommendations for contributors, I wanted to check if these are still of value (they are to me but maybe I am alone...). They are probably not that hard (I am currently working on 2. and have studied 1.) but I suspect I would need some help for a clean integration in the build process.

Matthieu
cyme.io

**Comment by mausch** [comment]
I don't know about Apple's specific frameworks but for general ARM support see https://github.com/facebookresearch/faiss/discussions/1577 .

If you want to use some specific BLAS implementations see the nix package definition : https://github.com/NixOS/nixpkgs/blob/457f28f6f85eea1b1671d62d2abe39297c9c90f7/pkgs/development/libraries/science/math/faiss/default.nix
